android/composeui/src/main/java/com/stadiamaps/ferrostar/composeui/views/InstructionsView.kt

```diff
@@ -36,23 +57,70 @@ fun InstructionsView(
+    initExpanded: Boolean = false,
+    contentBuilder: @Composable (VisualInstruction) -> Unit = {
+      ManeuverImage(it.primaryContent, tint = MaterialTheme.colorScheme.primary)
     }
 ) {
-  Column(
+  var isExpanded by remember { mutableStateOf(initExpanded) }
+  val screenHeight = LocalConfiguration.current.screenHeightDp.dp
+
+  Box(
       modifier =
           Modifier.fillMaxWidth()
-              .shadow(elevation = 5.dp, RoundedCornerShape(10.dp))
+              .heightIn(max = screenHeight)
+              .animateContentSize(animationSpec = spring(stiffness = Spring.StiffnessHigh))
               .background(theme.backgroundColor, RoundedCornerShape(10.dp))
-              .padding(8.dp)) {
-        ManeuverInstructionView(
-            text = instructions.primaryContent.text,
-            distanceFormatter = distanceFormatter,
-            distanceToNextManeuver = distanceToNextManeuver,
-            theme = theme,
-            content = content)
-        // TODO: Secondary instructions
+              .padding(16.dp)) {
+        LazyColumn(verticalArrangement = Arrangement.spacedBy(8.dp)) {
+          // Primary content
+          item {
+            ManeuverInstructionView(
+                text = instructions.primaryContent.text,
+                distanceFormatter = distanceFormatter,
+                distanceToNextManeuver = distanceToNextManeuver,
+                theme = theme) {
+                  contentBuilder(instructions)
+                }
+          }
+
+          // TODO: Secondary content
+
+          // Expanded content
+          if (isExpanded && remainingSteps != null && remainingSteps.count() > 1) {
+            item { HorizontalDivider(thickness = 1.dp) }
+            items(remainingSteps.drop(1)) { step ->
+              step.visualInstructions.firstOrNull()?.let { upcomingInstruction ->
+                Spacer(modifier = Modifier.height(8.dp))
+                ManeuverInstructionView(
+                    text = upcomingInstruction.primaryContent.text,
+                    distanceFormatter = distanceFormatter,
+                    distanceToNextManeuver = step.distance,
+                    theme = theme) {
+                      contentBuilder(upcomingInstruction)
+                    }
+                Spacer(modifier = Modifier.height(16.dp))
+                HorizontalDivider(thickness = 1.dp)
+              }
+            }
+          }
+
+          item {
+            Box(
+                modifier =
+                    Modifier.height(36.dp).fillMaxWidth().clickable { isExpanded = !isExpanded }) {
+                  Icon(
+                      if (isExpanded) {
+                        Icons.Default.KeyboardArrowUp
+                      } else {
+                        Icons.Default.KeyboardArrowDown
+                      },
+                      modifier = Modifier.align(Alignment.Center),
+                      contentDescription = "Show upcoming maneuvers")
+                }
```

Comment:
Getting there. I still think it the height has to be extremely small when not expanded. Maybe even so much so that we use an offset to overlay the pill on the padding to keep the view tight. As its own footer composable, maybe something like:

```fun
@Composable
fun InstructionsViewFooter(
  isExpanded: Boolean,
  iconTint: Color
) {
  val height: Dp = if (isExpanded) 36.dp else 8.dp

  Box(
      modifier =
          Modifier.fillMaxWidth()
              .height(height)
          // TODO: Clickable if not handled by the outer InstructionsView
  ) {
            if (isExpanded) {
              Icon(Icons.Default.KeyboardArrowUp,
                modifier = Modifier.align(Alignment.Center),
                contentDescription = "Show upcoming maneuvers",
                tint = iconTint)
            } else {
              Box(
                  modifier = Modifier.fillMaxWidth()
                      .height(6.dp)
                      .background(iconTint, RoundedCornerShape(6.dp)))
            }
          }
}
```

Apologies if this doesn't work (cargoNdk is still failing). But a centered pill with minimal height when not expanded is the goal. I don't think the full height down arrow button will acceptable for most app's UIs (too much wasted space).

As an aside. Google maps doesn't even have the full instrument list these days of some reason. Maybe why it's hard to find an android example.

Reply:
Thanks; will mess around with this some more. The reason it is larger currently is because the touch target was too small otherwise.

Reply:
Resolved by making the entire banner a click target _for expansion_. This lets us still use a thin pill and offsets to keep a slim view while keeping it usable.